bindings/dart/example/lib/logic.dart

```diff
@@ -200,3 +204,42 @@ class Outcome {
     return resultList;
   }
 }
+
+enum _BenchmarkStatsKind {
```

Comment:
Here I'll prefer to use inheritance 
```dart
abstract class BecnhmarkStats {
  static BecnhmarkStats none() { return _BecnhmarkStatsNone(); }
  static BecnhmarkStats pending(int iter) { return _BecnhmarkStatsIter(iter); }
  static BecnhmarkStats ready(Stats<num> stats) { ... }
}
```
We avoid `null` and we don't add too much boilerplate. It could be done with `freezed` but I don't think it is worth it.